src/lib/pages/backoffice/DocumentRequest/DocumentRequestDetails/DocumentRequestSteps/DocumentRequestStepPanels/helper.js

```diff
@@ -0,0 +1,41 @@
+import _cloneDeep from 'lodash/cloneDeep';
+import _set from 'lodash/set';
+import _get from 'lodash/get';
+
+/**
+ * @callback customizedValueFunction
+ * @param {*} value the value from the found property
+ * @param {object} document origin object that was passed to the extractAndCreate function
+ */
+
+/**
+ * @param {{mappings: [string , string, customizedValueFunction|undefined][], origin: object, destination: object}} args
+ *        all args are required
+ *
+ *
+ * @description
+ * Function that checks an array of arrays containing instructions in the following format [propertyToCheck, pathToPutPropertyValue, a function to customize the value of the property]
+ *@description
+ * it will check if the property (can be nested using the lodash dot syntax) exists in the origin object and if so it will set the property value on the chosen path in the destinationObject
+ */
+
+export const extractAndCreate = (args) => {
+  const { destination, mappings, origin } = args;
+  const clonedDestination = _cloneDeep(destination);
+
+  mappings.forEach((map) => {
```

Comment:
You are assuming that this is defined, maybe you want to a default value or a check before using it.

Reply:
agree 😄